ext/standard/php_image.h

```diff
@@ -54,4 +55,13 @@ PHPAPI int php_getimagetype(php_stream *stream, const char *input, char *filetyp
 
 PHPAPI char * php_image_type_to_mime_type(int image_type);
 
+/* Struct and function to be used for testing whether an image is AVIF */
+typedef struct {
+  php_stream * stream;
+  zend_string * data;
+  size_t data_pos;
+} php_gd_image_reader;
+
+PHPAPI int php_is_image_avif(php_gd_image_reader * reader);
```

Comment:
I'm not sure whether we want to expose this API. GD should probably call `php_getimagetype()` instead of having its own implementation. Of course, there is the issue of string vs. stream, but that could be resolved otherwise. Not even sure if we want to introduce `php_image_reader` – maybe we should just wrap a [memory stream](https://github.com/php/php-src/blob/master/main/streams/memory.c) around the string data?

Reply:
I agree - it would be nice if gd.c didn't replicate the image type detection logic in image.c, or vice versa. As a newbie, I didn't feel bold enough to make this change. Though, as you can see, I did insist on not replicating the more complex logic I'm introducing to detect AVIF images properly.

These memory streams are new to me. If I can use a memory stream to make a string look like a stream, I agree - that would be simpler than my new struct and new auxiliary function!

I'd defer to you here. Would it make sense to use the memory stream here, finish the AVIF support... and then create a followup PR to have gd.c simply use image.c's image type function?

Reply:
Yes, that sounds like a good plan! General [docs about streams](https://github.com/php/php-src/blob/master/docs/streams.md) are available; you likely need [php_stream_memory_open()](https://heap.space/xref/php-src/main/php_memory_streams.h?r=01b3fc03#31).